django/db/backends/mysql/operations.py

```diff
@@ -462,3 +464,15 @@ def on_conflict_suffix_sql(self, fields, on_conflict, update_fields, unique_fiel
             update_fields,
             unique_fields,
         )
+
+    def check_field_default_support(self, expression):
+        if not super().check_field_default_support(expression):
+            return False
+
+        # MySQL after 8.0.13 supports functions in defaults
+        if self.connection.features.supports_expression_defaults:
+            return True
+
```

Comment:
IMO this can be removed after resolving https://github.com/django/django/pull/16092#discussion_r1101104369.

Reply:
As the code currently is I think we still need this line - specifically, we want to special case that `Now` and `Value` are valid on old MySQL but all other expressions are not. This line is required to avoid ruling out all other expressions on later versions of MySQL.

I'm open to further refactors, but as I mentioned above I'm not seeing how that would work yet.